agent/ecr/factory_test.go

```diff
@@ -0,0 +1,37 @@
+// +build unit
+
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package ecr
+
+import (
+	"testing"
+
+	apicontainer "github.com/aws/amazon-ecs-agent/agent/api/container"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGetClientConfigEndpointOverride(t *testing.T) {
+	testAuthData := &apicontainer.ECRAuthData{
+		EndpointOverride: "ecr.us-west-2.amazonaws.com",
```

Comment:
Does this need an ACS model update?

Reply:
the EndpointOverride field is already in acs payload [here](https://github.com/aws/amazon-ecs-agent/blob/dev/agent/acs/model/ecsacs/api.go#L277)